commons/src/test/java/io/aiven/kafka/tieredstorage/commons/storage/s3/S3StorageConfigTest.java

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright 2023 Aiven Oy
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.tieredstorage.commons.storage.s3;
+
+import java.util.Map;
+
+import org.apache.kafka.common.config.ConfigException;
+
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.auth.AWSStaticCredentialsProvider;
+import com.amazonaws.auth.EnvironmentVariableCredentialsProvider;
+import com.amazonaws.regions.Regions;
+import com.amazonaws.services.s3.AmazonS3;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+class S3StorageConfigTest {
+
+    // Test scenarios
+    // - Minimal/Full configs
+    @Test
+    void minimalConfig() {
+        final String bucketName = "b1";
+        final Map<String, Object> configs = Map.of(S3StorageConfig.S3_BUCKET_NAME_CONFIG, bucketName);
+        final S3StorageConfig config = new S3StorageConfig(configs);
+        assertThat(config.bucketName()).isEqualTo(bucketName);
+    }
+
+    // - Credential provider scenarios
+    //   - Without provider
+    @Test
+    void configWithoutProvider() {
+        final String bucketName = "b1";
+        final String region = Regions.US_EAST_2.getName();
+        final String minioUrl = "http://minio";
+        final Map<String, Object> configs = Map.of(
+            S3StorageConfig.S3_BUCKET_NAME_CONFIG, bucketName,
+            S3StorageConfig.S3_REGION_CONFIG, region,
+            S3StorageConfig.S3_ENDPOINT_URL_CONFIG, minioUrl);
+        final S3StorageConfig config = new S3StorageConfig(configs);
+        assertThat(config.bucketName()).isEqualTo(bucketName);
+        assertThat(config.getString(S3StorageConfig.S3_REGION_CONFIG)).isEqualTo(region);
+        assertThat(config.getString(S3StorageConfig.S3_ENDPOINT_URL_CONFIG)).isEqualTo(minioUrl);
+        assertThat(config.getClass(S3StorageConfig.AWS_CREDENTIALS_PROVIDER_CLASS_CONFIG)).isNull();
```

Comment:
This checks don't really make much sense: they effectively test if the config framework of Kafka works correctly. We should get the client by `config.s3Client()` and then check all the necessary things on it (admittedly, with endpoint we'll have to access a private field; we can do this or skip). Same in other tests below.